### PR TITLE
test: prove that `AddressType` works

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

add tests to prove AddressType works for https://github.com/ApeWorX/ape/issues/1790

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: tests/functional/test_types.py [hunk 1]
```diff
@@ -147,3 +149,24 @@ def test_signature_from_rsv_and_vrs(signature):
...
+    # Show str works.
+    instance_str = MyModel(addr=owner.address)
+    assert instance_str.addr == owner.address
+
+    # Show hex bytes work.
+    instance_hex_bytes = MyModel(addr=HexBytes(owner.address))
+    assert instance_hex_bytes.addr == owner.address
+
+    # Show raw bytes work.
+    instance_bytes = MyModel(addr=bytes.fromhex(owner.address[2:]))
```
Comment (antazoey) [inline tests/functional/test_types.py:167]: @Aviksaikat This is the case you are in.  If you are using raw bytes, make sure it looks the same as doing `bytes.fromhex()` on the hex str _without_ the 0x prefix... or even simpler, you can just use the hex str directly!

thank you

Comment (fubuloubu) [reply tests/functional/test_types.py:167]: In general, we are using Pydantic v2 now, which is less forgiving on various "implicit conversions" that may have been happening before

Comment (Aviksaikat) [reply tests/functional/test_types.py:167]: Cool. :+1:

Comment (antazoey) [reply tests/functional/test_types.py:167]: Bad things can also happen if we try to handle bytes encoded hex strings like `b"0x123"`, the mixing of the formats leads to ambiguity.

**Pull request merged**